client/x/evmengine/keeper/keeper_internal_test.go

```diff
@@ -292,81 +287,43 @@ func TestKeeper_isNextProposer(t *testing.T) {
 	t.Parallel()
 	height := int64(1)
 	tests := []struct {
-		name       string
-		args       args
-		want       bool
-		wantHeight uint64
-		wantErr    bool
+		name    string
+		args    args
+		want    bool
+		wantErr bool
 	}{
 		{
-			name: "is next proposer",
+			name: "not proposer",
 			args: args{
-				height:  height,
-				current: 0,
-				next:    1,
-				header: func(height int64, address []byte) cmtproto.Header {
-					return cmtproto.Header{Height: height, ProposerAddress: address}
+				height:       height,
+				incMoreTimes: 9,
```

Comment:
This argument seems to cause the test to fail sometimes. Why don't we just take a bool whether it's nextProposer or not and set the validator address to something other than the proposer if it's not?

Reply:
Good point, I've done that in `abci_internal_test.go` but missed this one. Resolved in e35ea5a, if not next proposer, then set `nxtAddr` (next proposer) to zero address (`0x00...00`)